test/test_mps.py

```diff
@@ -139,8 +139,8 @@ def mps_ops_grad_modifier(ops):
         'round': [torch.float16],
 
         # atomic operation in backward pass
-        '_unsafe_masked_index': [torch.int8, torch.uint8, torch.float16, torch.int16, torch.int64],
-        '_unsafe_masked_index_put_accumulate': [torch.int8, torch.uint8, torch.float16, torch.int16, torch.int64],
+        '_unsafe_masked_index': [torch.bool, torch.int8, torch.uint8, torch.float16, torch.int16, torch.int64],
+        '_unsafe_masked_index_put_accumulate': [torch.bool, torch.int8, torch.uint8, torch.float16, torch.int16, torch.int64],
```

Comment:
How do integer types end up in the backward pass?

Reply:
I'm not sure I understand the question. MPS accumulate supports only `float32` and `int32`

Reply:
The comment says "atomic operations in the backward pass" but you can't get the gradient of an integer tensor. So something isn't right here.

Reply:
Thanks. I misread the test failures. Ready for a review now.